app/Repositories/ReportbackRepository.php

```diff
@@ -50,4 +89,57 @@ public function create($data)
 
         return $reportback;
     }
+
+    /**
+     * Add a new item to a reportback.
+     *
+     * @param  \Rogue\Models\Reportback $reportback
+     * @param  array $data
+     *
+     * @return \Rogue\Models\Reportback
+     */
+    public function addItem($reportback, $data)
+    {
+        if (isset($data['file'])) {
+            // @todo - this part right here might actually belong in the service class now that i think about it.
+            $data['file_url'] = $this->aws->storeImage($data['file'], $data['file_id']);
+
+            $reportback->items()->create(array_only($data, ['file_id', 'file_url', 'caption', 'status', 'reviewed', 'reviewer', 'review_source', 'source', 'remote_addr']));
+        }
+
+        return $reportback;
+    }
+
+    /*
+     * Get a user's reportback based on their drupal id or their northstar id.
+     *
+     * @param string|int $campaignId
+     * @param string|int $campaignRunId
+     * @param string|int $userId
+     * @param string $type
+     *
+     * @return \Rogue\Models\Reportback|null
+     */
+    public function getReportbackByUser($campaignId, $campaignRunId, $userId, $type = 'northstar_id')
+    {
+        try {
+            if ($type === 'drupal_id') {
+                $reportback = Reportback::where([
```

Comment:
Not sure if it's more or less readable, but these two cases could be combined & enforced with a guard:

```php
if (! in_array($type, ['northstar_id', 'drupal_id']) {
    throw new \Exception('Invalid user ID type provided.');
}

return Reportback::where([
    $type => $userId,
    'campaign_id' => $campaignId,
    'campaign_run_id' => $campaignRunId,
])->firstOrFail();
```